### PR TITLE
token-voting

### PR DESCRIPTION
**Pull request by novaknole**
## Description

since I added the check in token factory, it started to fail tests ! really don't have time to be fixing the token factory tests since it's not priority and token factory anyways mightn't be finished and it's mostly for future ! so, I'd not worry about skipped test now and we track this in february just in case to make it green.

Task: [ID]()

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: packages/contracts/contracts/factory/TokenFactory.sol [hunk 1]
```diff
@@ -49,6 +49,10 @@ contract TokenFactory {
     /// @param token GovernanceWrappedERC20 token address
     event WrappedToken(GovernanceWrappedERC20 token);
 
+    /// @notice Thrown if token address is not ERC20.
+    /// @param token The token address
+    error TokenNotERC20(address token, bytes data);
```
Comment (brickpop) [inline packages/contracts/contracts/factory/TokenFactory.sol:54]: Aren't we we're supporting ERC20, 721 and 1155? Or is this only in regards of the DAO deposit/transfer?

File: packages/contracts/contracts/voting/token/TokenVotingSetup.sol [hunk 2]
```diff
@@ -296,13 +288,13 @@ contract TokenVotingSetup is PluginSetup {
         return token.getSupportedInterfaces(interfaceIds);
     }
 
-    /// @notice unsatisfiably determines if contract is ERC20..
-    /// @dev it's important to check first whether token is a contract.
-    /// @param token address
+    /// @notice Unsatisfiably determines if contract is ERC20..
```
Comment (brickpop) [inline packages/contracts/contracts/voting/token/TokenVotingSetup.sol:291]:
```suggestion
    /// @notice Unsatisfiably determines if the contract is an ERC20 token.
```

File: packages/contracts/contracts/factory/TokenFactory.sol (hunk 1 above)
Comment (novaknole) [reply packages/contracts/contracts/factory/TokenFactory.sol:54]: We don't support it yet.. Maybe in next versions.

File: packages/contracts/test/voting/token-voting-setup.ts [hunk 3]
```diff
@@ -40,7 +40,7 @@ const UPGRADE_PERMISSION_ID = ethers.utils.id('UPGRADE_PLUGIN_PERMISSION');
 const EXECUTE_PERMISSION_ID = ethers.utils.id('EXECUTE_PERMISSION');
 const MINT_PERMISSION_ID = ethers.utils.id('MINT_PERMISSION');
 
-describe('TokenVotingSetup', function () {
+describe.only('TokenVotingSetup', function () {
```
Comment (mathewmeconry) [inline packages/contracts/test/voting/token-voting-setup.ts:43]: remove `only`

**Review by mathewmeconry — CHANGES_REQUESTED** [review]

File: packages/contracts/test/voting/token-voting-setup.ts (hunk 3 above)
Comment (novaknole) [reply packages/contracts/test/voting/token-voting-setup.ts]: DONE

**Review by mathewmeconry — DISMISSED** [review]

File: packages/contracts/contracts/voting/token/TokenVotingSetup.sol [hunk 4]
```diff
@@ -40,6 +38,10 @@ contract TokenVotingSetup is PluginSetup {
     /// @notice The address of the `GovernanceWrappedERC20` base contract.
     address public immutable governanceWrappedERC20Base;
 
+    /// @notice The token settings structure holds information on whether to deploy a new contract or utilize the provided address.
+    /// @param addr The token address, if non-zero, wraps it with GovernanceWrappedERC20, otherwise, deploys GovernanceERC20.
+    /// @param name The token name. Only relevant if token address is zero.
+    /// @param name The token symbol. Only relevant if token address is zero.
```
Comment (Michael-A-Heuer) [inline packages/contracts/contracts/voting/token/TokenVotingSetup.sol:44]:
```suggestion
    /// @notice The token settings struct.
    /// @param addr The token address. If this is `address(0)`, a new `GovernanceERC20` token is deployed. If not, the existing token is wrapped as an `GovernanceWrappedERC20`.
    /// @param name The token name. This parameter is only relevant if the token address is `address(0)`.
    /// @param name The token symbol. This parameter is only relevant if the token address is `address(0)`.
```

File: packages/contracts/contracts/voting/token/TokenVotingSetup.sol [hunk 5]
```diff
@@ -132,7 +130,7 @@ contract TokenVotingSetup is PluginSetup {
                 // it's simply ERC20 which gets checked by _isERC20
                 // Currently, not a satisfiable check..
```
Comment (Michael-A-Heuer) [inline packages/contracts/contracts/voting/token/TokenVotingSetup.sol:131]: Don't we want to remove the `// Currently, not a satisfiable check..`?

Comment (novaknole) [reply packages/contracts/contracts/voting/token/TokenVotingSetup.sol:131]: Let's leave it. It's true and helps other devs get the point.

Comment (brickpop) [inline packages/contracts/contracts/voting/token/TokenVotingSetup.sol:131]: Tiny silly change that does nothing when you get it right, but makes things feel careless when it's not polished. 

```suggestion
                // Currently, not a satisfiable check.
```

**Review by Michael-A-Heuer — APPROVED** [review]

**Pull request merged**